### PR TITLE
Refactored scripts organization

### PR DESCRIPTION
**Pull request by AndreasLrx**
# Description

Moved all scripts in appropriate subfolders.

Requires #75 

## Type of change

- [x] Refactoring

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by EnzoDrapron — APPROVED** [review]

File: Assets/Scripts/Generation/Room.cs [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+    // Room bounding box (minimum box where all parts fits in)
+    private RectInt roomBoundingBox;
+    // Boundaries of the camera (list of points)
+    public List<Vector2> roomBoundaries;
+
+    // Associative double dimension array representing the cells
+    private List<List<Cell>> grid;
+
+
```
Comment (EnzoDrapron) [inline Assets/Scripts/Generation/Room.cs:44]: Delete new line

**Review by cley44 — APPROVED** [review]

File: Assets/Scripts/Generation/RoomGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+        {
+            Direction.top => new Vector2(position.x, position.y + 1),
+            Direction.right => new Vector2(position.x + 1, position.y),
+            Direction.bottom => new Vector2(position.x, position.y - 1),
+            Direction.left => new Vector2(position.x - 1, position.y),
+            _ => position,
+        };
+    }
+
+    Direction NextDirection(Direction direction)
```
Comment (EnzoDrapron) [inline Assets/Scripts/Generation/RoomGenerator.cs:79]: add private

File: Assets/Scripts/Generation/RoomGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+    {
+        if (direction == Direction.left)
+            return 0;
+        else
+            return direction + 1;
+    }
+
+    Vertice BorderDirectionToVertice(Direction direction, Vector2 partPos)
```
Comment (EnzoDrapron) [inline Assets/Scripts/Generation/RoomGenerator.cs:87]: add private

File: Assets/Scripts/Generation/RoomGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+        while (innerWallCount * innerWallLength > partSize.x * partSize.y * 0.8)
+        {
+            if (innerWallCount > 1)
+                innerWallCount--;
+            if (innerWallLength > 1)
+                innerWallLength--;
+        }
+    }
+
+    Vector2? GetInnerWallStartPosition(int depth = 0)
```
Comment (EnzoDrapron) [inline Assets/Scripts/Generation/RoomGenerator.cs:271]: add private

File: Assets/Scripts/Generation/RoomGenerator.cs [hunk 5]
```diff
@@ -0,0 +1,422 @@
...
+    {
+        RoomSetup();
+        // This might fail and calls a room regeneration
+        if (!GenerateInnerWalls())
+            return false;
+        room.InstantiateTiles();
+        return true;
+    }
+
+    void SetupCameraBoundaries()
```
Comment (EnzoDrapron) [inline Assets/Scripts/Generation/RoomGenerator.cs:334]: add private

File: Assets/Scripts/Generation/RoomGenerator.cs [hunk 6]
```diff
@@ -0,0 +1,422 @@
...
+    // Boundaries of the camera (list of vertices)
+    private List<Vertice> cameraBoundaries;
+    // List of empty positions
+    private List<Vector2> emptyPositions;
+
+    // Generated room object
+    private Room room;
+
+    // Helper methods
+    Vector2 MoveInDirection(Vector2 position, Direction direction)
```
Comment (EnzoDrapron) [inline Assets/Scripts/Generation/RoomGenerator.cs:67]: add private

File: Assets/Scripts/Generation/Room.cs (hunk 1 above)
Comment (EnzoDrapron) [inline Assets/Scripts/Generation/Room.cs:44]: delete new line

**Review by EnzoDrapron — CHANGES_REQUESTED** [review]

**Review by cley44 — APPROVED** [review]

**Pull request merged**